Add more content to the Basic Auth Howto document (#29495)
quarkusio/quarkus: docs/src/main/asciidoc/security-basic-authentication-howto.adoc

quarkus.http.auth.basic=true
----
 

> The easiest way to configure user credentials required for a Basic authentication to work is to configure the user name, secret and roles directly in `application.properties`, see the xref:security-testing.adoc#configuring-user-information[Configuring User Information] section for more information.

Reviewer: ```suggestion
An easy way to configure the required user credentials for Basic authentication to work is to configure the user name, secret, and roles directly in the `application.properties` file.

.Example of Basic authentication properties

_[insert example snippet in here]_

For more information, see xref:security-testing.adoc#configuring-user-information[Configuring User Information] in the "Security Testing" guide.
```

> The easiest way to configure user credentials required for a Basic authentication to work is to configure the user name, secret and roles directly in `application.properties`, see the xref:security-testing.adoc#configuring-user-information[Configuring User Information] section for more information.

Reviewer: @sberyozkin - 2 things:

1. Sentence contains 35 words.
The above-suggested change reduces the sentence length. To improve readability and clarity, the max sentence length recommendation for Quarkus contributors (defined in doc-reference.adoc) is 32 words, with the exception of the first sentence (preamble), which must be less than 26 to fit/look good on the index page tile.
[When the Vale linter is fully implemented, it will detect and flag this btw...]. 

2. References: We try to use a consistent approach to different types of cross-references across all of the docs. The above is widely used across the product docs and will support single-sourcing too. We can make the contributor guide a bit clearer here and add examples. I can take that action ...

Author: Thanks for the explanation @michelle-purcell

> However, configuring the user name, secret and roles in `application.properties` is only suitable for testing and instead, using a database to store this information is recommended.

Reviewer: I think this statement could be stronger. Suggestion :arrow_down: WDYT?

```suggestion
[IMPORTANT]
====
Configuring user names, secrets, and roles in the `application.properties` file is only suitable for testing scenarios. If you are securing an application for production, always use a database to store this information.
====
```

> For a Basic authentication configuration walk-through that uses `JPA` to store user credentials in a database, see the xref:security-basic-authentication-tutorial.adoc[Secure a Quarkus application with Basic authentication] guide.

Reviewer: ```suggestion
To walk through how to configure Basic authentication together with JPA for storing user credentials in a database, see the xref:security-basic-authentication-tutorial.adoc[Secure a Quarkus application with Basic authentication] tutorial.
```

Reviewer: @sberyozkin - I know you didn't touch this sentence in this PR but while we are here, should we simplified the phrasing as per above :arrow_up: ?  Thanks .
